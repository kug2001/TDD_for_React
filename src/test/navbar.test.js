import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/navbar';

describe('habits에 대한 테스트 코드', () => {

  it('스냅샷 테스트', () => {
    const component = renderer.create(<Navbar />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('navbar에 테스트 코드', () => {
    it('total 카운트 0표기 확인', () => {
      render(<Navbar totalCount={0} />);
      const count = screen.getByTestId('total-count');
      expect(count.textContent).toBe("0");
    });
    it('total 카운트 99 표기 확인', () => {
      render(<Navbar totalCount={99} />);
      const count = screen.getByTestId('total-count');
      expect(count.textContent).toBe("99");
    });
  })
})