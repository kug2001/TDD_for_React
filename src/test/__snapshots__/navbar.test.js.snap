// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habits에 대한 테스트 코드 스냅샷 테스트 1`] = `
<nav
  className="navbar"
>
  <i
    className="navbar-logo fas fa-leaf"
  />
  <span>
    Habit Tracker
  </span>
  <span
    className="navbar-count"
    data-testid="total-count"
  />
</nav>
`;
